src/plugins/intel_gpu/src/plugin/program.cpp

```diff
@@ -318,7 +318,7 @@ std::shared_ptr<cldnn::program> Program::BuildProgram(const std::vector<std::sha
         options.set_option(cldnn::build_option::graph_dumps_dir(m_config.graph_dumps_dir));
     }
     for (const auto& op : ops) {
-        if (op->is_dynamic()) {
+        if (op->is_dynamic() || strncmp(op->get_type_name(), "MatMul", 6) == 0) {
```

Comment:
I guess this is for temporal change, but this wont' work for many MatMul static tests, because there will be side effects for many operations for static case tests.